quipucords/api/source/serializer.py

```diff
@@ -606,3 +621,33 @@ class Meta:
             "credentials",
             "most_recent_connect_scan",
         )
+
+    @transaction.atomic
+    def create(self, validated_data):
+        """Create a V2 source."""
+        instance = super().create_base(validated_data)
+        return instance
+
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        """Update a V2 source."""
+        ssl_options = {}
+        for ssl_option in self.SSL_OPTIONS:
+            value = validated_data.pop(ssl_option, None)
+            if value is not None:
+                ssl_options[ssl_option] = value
+        instance = super().update_base(instance, validated_data)
+        if ssl_options:
+            source_type = instance.source_type
+            self.validate_opts(ssl_options, source_type)
+            self.update_options(ssl_options, instance)
+            instance.save()
+        return instance
+
+
+# Let's define the SourceSerializer based on the API_VERSION
+match API_VERSION:
+    case 2:
+        SourceSerializer = SourceSerializerV2
+    case _:
+        SourceSerializer = SourceSerializerV1
```

Comment:
I'm a bit confused here. Won't we use `request.version` to find the version at runtime? why do we need API_VERSION constant? 🤔

Reply:
gotcha. It's a nice way to have a smaller diff.

Reply:
leave it as is, :shipit: